zerver/tests/test_scheduled_messages.py

```diff
@@ -314,6 +323,14 @@ def test_delivery_type_reminder_failed_to_deliver_scheduled_message_unknown_exce
             )
             self.assertTrue(scheduled_message.failed)
 
+            # Verify that the user was sent a message informing them about
+            # the failed scheduled message.
+            realm = scheduled_message.sender.realm
+            msg = most_recent_message(scheduled_message.sender)
+            self.assertEqual(msg.recipient.type, msg.recipient.PERSONAL)
+            self.assertEqual(msg.sender_id, self.notification_bot(realm).id)
+            self.assertIn("Internal server error", msg.content)
```

Comment:
I like us to have explicit test coverage the "the message failed because the user was deactivated shouldn't generate a notification bot message".

Reply:
Added explicit text coverage for both deactivated cases: realm and user.